### ISSUE TITLE
QUICLossDetector crashes on loss detection timeout

### ISSUE THREAD

**Issue by maskit** [apache/trafficserver#7044]
```
Fatal: QUICLossDetector.cc:450: failed assertion `this->_is_client_without_one_rtt_key()`
```
```
traffic_server: received signal 6 (Aborted)
traffic_server - STACK TRACE: 
/opt/ats/lib/libtscore.so.10(_Z20signal_crash_handleriP9siginfo_tPv+0x18)[0x7f114cbced18]
/opt/ats/bin/traffic_server(_Z19crash_logger_invokeiP9siginfo_tPv+0x168)[0x78a5b0]
/lib64/libpthread.so.0(+0x12080)[0x7f114b269080]
/lib64/libc.so.6(gsignal+0x10b)[0x7f114a38beab]
/lib64/libc.so.6(abort+0x123)[0x7f114a3765b9]
/opt/ats/lib/libtscore.so.10(_Z11ink_warningPKcz+0x0)[0x7f114cb73396]
/opt/ats/lib/libtscore.so.10(+0x202fc3)[0x7f114cb6afc3]
/opt/ats/bin/traffic_server(_ZN16QUICLossDetector26_on_loss_detection_timeoutEv+0x473)[0x100f46f]
/opt/ats/bin/traffic_server(_ZN16QUICLossDetector13event_handlerEiP5Event+0x19a)[0x1009558]
/opt/ats/bin/traffic_server(_ZN12Continuation11handleEventEiPv+0x19e)[0x796936]
/opt/ats/bin/traffic_server(_ZN7EThread13process_eventEP5Eventi+0x331)[0xf6d633]
/opt/ats/bin/traffic_server(_ZN7EThread15execute_regularEv+0x720)[0xf6e476]
/opt/ats/bin/traffic_server(_ZN7EThread7executeEv+0x2e9)[0xf6f0cb]
/opt/ats/bin/traffic_server[0xf6b9a0]
/lib64/libpthread.so.0(+0x7594)[0x7f114b25e594]
/lib64/libc.so.6(clone+0x3f)[0x7f114a44ef4f]
```

The assert is new and only on quic-latest (draft-29). This doesn't happen on 9.0.0.

**Comment by maskit** [comment]
A similar assertion failure was pointed out on the Slack, and an update was made to the pseudo code.
Following the latest pseudo code may solve the issue.

**Comment by maskit** [comment]
We need to fix this because 9.0 has draft-29 support.
